components/operator/internal/resources/benthos/controller.go

```diff
@@ -93,8 +93,14 @@ func createDeployment(ctx Context, stack *v1beta1.Stack, b *v1beta1.Benthos) err
 		return errors.Wrap(err, "searching elasticsearch configuration")
 	}
 
+	serviceAccountName, err := settings.GetAWSServiceAccount(ctx, stack.Name)
+	if err != nil {
+		return err
+	}
+
+	awsIAMEnabled := serviceAccountName != ""
 	var resourceReference *v1beta1.ResourceReference
-	if secret := elasticSearchURI.Query().Get("secret"); secret != "" {
+	if secret := elasticSearchURI.Query().Get("secret"); !awsIAMEnabled && secret != "" {
```

Comment:
So you're not creating any secret if iam is enabled, am I right ?

Reply:
Yes !